src/api/spec/models/attrib_spec.rb

```diff
@@ -1,19 +1,106 @@
 require 'rails_helper'
 
-RSpec.describe Attrib do
+RSpec.describe Attrib, :type => :model do
   let(:attribute) { create(:attrib, project: create(:project)) }
   let(:project) { create(:project) }
   let(:package) { create(:package) }
 
+  context "#fullname" do
+    it { expect(attribute.fullname).to eq("#{attribute.namespace}:#{attribute.name}") }
+  end
+
+  context "#container" do
+    context "attribute with project" do
+      it { expect(attribute.container).to eq(attribute.project) }
+    end
+
+    context "attribute with package" do
+      let(:attribute_with_package) { create(:attrib, package: package) }
+
+      it { expect(attribute_with_package.container).to eq(package) }
+    end
+  end
+
   context "#container=" do
-    it "sets the project" do
-      attribute.container = project
-      expect(attribute.container).to be(project)
+    context "assigning a project" do
+      before do
+        attribute.container = project
+      end
+
+      it { expect(attribute.container).to be(project) }
+      it { expect(attribute.project).to be(project) }
+      it { expect(attribute.package).to be_nil }
+
+      context "and then assigning a package" do
```

Comment:
> > I would expect that assigning a package and then a project results in a project. But no. For this last case it results in a package.
> 
> Thats because the `container` method first checks if there is a package set and returns that if it exists, then it checks if there is a project set.

That is the reason this test was created.

> You are testing two different methods here, one is the `container=` method and the other is the `container` method. IMO The tests would make sense if they were split into two describe blocks, one for `container=` and one for `container`. Then you don't have to use `context 'and then...'`.
> 
> https://gist.github.com/evanrolfe/586f0a529bb1a0ae5ef147c133ff6304

Changed `context` to `describe`. It was wrong.

In your gist code I don't see a test of comparing the values `container`, `project` and `package` methods returns after an assignment of a package and then a project. Should I drop that test then?